_posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc

```diff
@@ -0,0 +1,549 @@
+of various groups and roles, and the configuration of multi-factor authentication (MFA) devices.
+
+=== Logging
+_Number of recommendations: 9_
+
+AWS has a variety of logging, monitoring, and auditing features, and the Benchmark has recommendations for
+several of them:
+
+* link:https://aws.amazon.com/cloudtrail/[AWS CloudTrail] tracks user activity and API usage
+* link:https://aws.amazon.com/config/[AWS Config] records and evaluates resource configurations
+* link:https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs.html[VPC Flow Logs] capture network traffic information
+  in VPCs
+
+AWS has several other logging related features that are not covered directly by the Benchmark. For example,
+the primary log ingestion and query service, link:https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/WhatIsCloudWatchLogs.html[Amazon CloudWatch
+Logs], is integrated with many other AWS services. The Benchmark does not contain any recommendations
+specifically for CloudWatch Logs, though many recommendations do make mention of it.
+
+=== Monitoring
+_Number of recommendations: 14_
+
+Monitoring is an overloaded term in the industry. In the context of the AWS Foundations Benchmark, the
+monitoring section is exclusively about monitoring for specific API calls using the CloudTrail service paired
+with link:https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/MonitoringLogData.html[CloudWatch Logs
+filter metrics]. Each recommendation in this section spells out a specific filter and an associated alarm.
+
+=== Networking
+_Number of recommendations: 4_
+
+The Benchmark is uncomfortably light on networking, considering its central role in the security of any
+distributed system. The recommendations merely limit traffic from the zero network (`0.0.0.0/0`) and
+restrict routing for VPC peering connections.
+
+[[production_grade_design]]
+== Production-grade design
+
+Gruntwork offers battle-tested infrastructure as code modules to help you create production grade
+infrastructure in a fraction of the time it would take to develop from scratch. Each of the IaC modules are
+"compliance-ready"; they are mostly unopionated by default, but they can be configured for compliance with the
+right configuration.
+ +
+ +
+To make it easy to achieve compliance, the Gruntwork Compliance modules are "wrappers" around the core,
+unopionated modules in the Infrastructure as Code Library. The wrappers call the core modules with
+configuration values that are compliant with the AWS Foundations Benchmark. You can use the wrapper modules by
+creating a module of your own (this can be considered a second wrapper) and using the compliance module as the
+`source`. Optionally, you can also use `terragrunt` to call your module, thus creating a chain of IaC
+modules.
+ +
+ +
+
+.Nested wrapper modules help to avoid repetitive code and minimize the amount of extra work needed to achieve compliance.
+image::/assets/img/guides/cis-compliance/wrappers.png[]
```

Comment:
I forget: was the goal to have the ability to directly call the compliance modules from `infrastructure-live` or is `infrastructure-modules` wrapper always necessary? I ask because this visual and explanation makes it seem that the infra modules wrapper is always necessary.

Note sure if there is a better way to visualize + describe this fact though...

Reply:
Yes! As we discussed separately, users can call the compliance modules from `infrastructure-modules`, `infrastructure-live`, or have both. I'll update the text to explain this.